### PR TITLE
Store per-song skip, hint count, time played, and time to guess

### PR DESCRIPTION
**Pull request by taahamahdi**
* Rename `song_guess_count` to `song_metadata`

### CODE REVIEW

File: src/structures/game_session.ts [hunk 1]
```diff
@@ -215,6 +221,7 @@ export default class GameSession {
         this.gameRound = null;
 
         gameRound.interactionMarkAnswers(guessResult.correctGuessers?.length);
+        const guessSpeed = Date.now() - gameRound.startedAt;
```
Comment (Brainicism) [inline src/structures/game_session.ts:224]: If you move it out here, this means that skips count as well?

Comment (Brainicism) [reply src/structures/game_session.ts:224]: Should we rename this?

**Review by Brainicism — CHANGES_REQUESTED** [review]

File: src/structures/game_session.ts (hunk 1 above)
Comment (taahamahdi) [reply src/structures/game_session.ts:224]: Oops, I'll rename it to `timePlayed`

**Review by Brainicism — APPROVED** [review]

File: src/migrations/20211216025236_song_guess_count_to_song_metadata.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+exports.up = function (knex) {
+    return knex.schema
+        .renameTable("song_guess_count", "song_metadata")
+        .table("song_metadata", (table) => {
+            table.renameColumn("correct_guesses", "correct_guesses_legacy");
```
Comment (Brainicism) [inline src/migrations/20211216025236_song_guess_count_to_song_metadata.js:5]: Don't we need to undo this rename in the exports.down?

**Review by Brainicism — CHANGES_REQUESTED** [review]

File: src/migrations/20211216025236_song_guess_count_to_song_metadata.js (hunk 2 above)
Comment (taahamahdi) [reply src/migrations/20211216025236_song_guess_count_to_song_metadata.js:5]: It falls back to the existing columns (correct_guesses and rounds_played will already exist)

**Review by Brainicism — APPROVED** [review]

**Pull request merged**